File: apps/go/internal/middleware/cors.go
package middleware

import (
	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
)

func Cors() gin.HandlerFunc {
	return cors.New(cors.Config{
		AllowOrigins: []string{"http://localhost:5173"},
		AllowMethods: []string{"GET", "POST", "PUT", "PATCH", "DELETE", "HEAD", "OPTIONS"},
		AllowHeaders: []string{"Origin", "Content-Type", "Content-Length", "Authorization"},
		// browsers withhold non-safelisted response headers from JS unless named here
		ExposeHeaders: []string{"Link", "X-Total-Count"},
	})
}

File: apps/js/src/middleware/cors.ts
import baseCors from "cors";
import type { RequestHandler } from "express";

const configured = baseCors({
	origin: "http://localhost:5173",
	methods: ["GET", "POST", "PUT", "PATCH", "DELETE", "HEAD", "OPTIONS"],
	allowedHeaders: ["Origin", "Content-Type", "Content-Length", "Authorization"],
	// browsers withhold non-safelisted response headers from JS unless named here
	exposedHeaders: ["Link", "X-Total-Count"],
});

// gin-contrib/cors answers a request without an Origin untouched; this matches it
export const cors: RequestHandler = (req, res, next) => {
	if (req.get("origin") === undefined) {
		next();
		return;
	}

	configured(req, res, next);
};
